### PR TITLE
Consolidate aggregators

### PR DESCRIPTION
**Pull request by aarangop**
Replaced the old functions with the parameterized generate_podcast_feed.

- Replaced the use of feedconfig parser by ElementTree to replace the formatted strings from the old code
- Would close: 
- #1 
- #2 
- #4 
- #5 
- #6 
- #8

### CODE REVIEW

**Review by maximumpeaches — COMMENTED** [review]
Hey can you add a commit or two with a test that illustrates how you plan to use the fixtures and the XML files from BeyondWords? Then I can finish my review.

File: functions/storage.py [hunk 1]
```diff
@@ -8,10 +9,25 @@ class StorageInterface:
...
+        self.output_feed_filename = f'{output_feed_filename_base}.xml'
+
+    def read_history_titles(self) -> List[str]:
         raise NotImplementedError()
 
-    def read_removed_authors(self):
+    def write_history_titles(self, history_titles: List[str]) -> int:
+        raise NotImplementedError()
+
+    def save_podcast_feed(self, feed: str):
```
[the remark this answers is not in the document]
Comment (aarangop) [reply functions/storage.py:27]: Will stick to read/write

File: tests/create_testing_snapshots.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import xml.etree.ElementTree as ET
+from datetime import datetime, timedelta, timezone
+
+import requests
+
+
+def save_beyondwords_snapshot(n_days=7, output_filename: str = None, reference_date=datetime.now(tz=timezone.utc),
```
[the remark this answers is not in the document]
Comment (aarangop) [reply tests/create_testing_snapshots.py:7]: This file should be used only to save an xml file from the feed with a reduced number of entries and a reference date. The reference date can be used to mock datetime.now() in the tests. The xml file will later be injected to the tests to check the output from the podcast_feed_generator. I'll move the file somewhere else and keep only tests and test-related code in the tests directory.

**Comment by aarangop** [comment]
The latest commit has a simple test that checks that the produce feed has at least one entry. 

For this to work without downloading the feed from beyondwords I implemented several mocks/patches. 

Datetime.now() freezes during the test to the reference_date which is provided by the beyondwords_snapshot.xml file.

The get_feed_tree_from_source is patched so it returns the root of the xml tree of the beyondwords_snapshot.xml. If the file doesn't exist, it will be created.

I also applied some patches to the read/write methods of the LocalStorage class, since the get_feed_generator saves the new history file. If the history is updated and the test is re-run, then the test will fail because all entries would now be in the history.

**Comment by aarangop** [comment]
Also for the alignment forum I wrote the function get_af_*_config, in order to load the config directly to the tests, instead of having to re-create the config in the tests. Let me know if you think this is a good/bad design so we can revert it or apply it to the rest of the aggregators. This allows us to use the exact config in the tests, but also forces us to mock things, for instance in order to avoid messing up the history.

**Comment by maximumpeaches** [comment]
> Also for the alignment forum I wrote the function get\_af\_\*\_config, in order to load the config directly to the tests, instead of having to re-create the config in the tests. Let me know if you think this is a good/bad design so we can revert it or apply it to the rest of the aggregators. This allows us to use the exact config in the tests, but also forces us to mock things, for instance in order to avoid messing up the history.

Yeah so in general when writing code you want to follow the DRY principle, but in tests you want to follow the DAMP principle. You can google "DAMP DRY tests" to see what these mean. We should just copy the config values to the tests themselves. It's okay that we're repeating code in our tests.

Generally a really good pattern I've seen for tests at Google is to have a method in tests which creates a default value, and then you can set whatever matters to that specific test in the test itself. So if your test is confirming that the posts with "AF - " are included, then you would use the default value and just replace the "AF - " in the default value. But for this test you wouldn't need to replace the GCP bucket "rssfile" because the test isn't confirming whether it was written to the right bucket. This helps a lot to make the test readable because when you read the test, you can see exactly what matters to the test, and it isn't cluttered by a lot of setup code that has nothing to do with what's being tested.

So I'd propose having a method in our tests which returns a default FeedGeneratorConfig, and then override specific values in that default object in each test. For practical purposes, it's totally fine if you want to do this as part of a separate PR... maybe in an ideal world you'd do it as part of this PR, but it may be a little easier to do it in a separate one. Up to you.

I'm not entirely sure how the fixtures in conftest.py will be used, but I think it's okay to just merge the PR and we may have to go back and change a few things later, but they won't be major things or difficult to change. I see that you're mocking the file and network dependencies, and that makes sense to do.

**Comment by maximumpeaches** [comment]
I'm going to approve the PR but maybe you can copy the config values to a method in our tests which returns a default value in this PR before merging? (Or would I have to come back and approve again if you change a file? If so then feel free to just merge it and we can work on that in a separate PR. The PRs don't have to be perfect or something - more important we make forward progress.) Thanks for your work! It's coming along!

**Pull request merged**